Initial implementation (#1)
projectsyn/component-nfs-subdir-external-provisioner: docs/modules/ROOT/pages/how-tos/aws-efs.adoc

== Amazon Web Services EFS

AWS EFS provides filesystem based storage access based on NFS.
Multiple pods on different nodes can access the same storage (RWX).

> An [CSI driver for EFS](https://github.com/kubernetes-sigs/aws-efs-csi-driver) is being built by AWS, but as of today, "Currently only static provisioning is supported".

Reviewer: ```suggestion
A [CSI driver for EFS](https://github.com/kubernetes-sigs/aws-efs-csi-driver) is being built by AWS, but as of February 2021, "only static provisioning is supported".
```

This is not the user experience we want to provide and therefore the Kubernetes NFS Subdir External Provisioner is the most practicable operator currently available.

=== Prerequisites

> The operator requires an already set up EFS.

Reviewer: ```suggestion
The operator requires that EFS is already setup.
```

> This can be created manually in the AWS UI or better through a management tool like Terraform.

Reviewer: ```suggestion
It's up to the operator how to setup EFS.
The provisioner works with a manually setup EFS volume through the AWS UI, or an EFS volume which was provisioned with a management tool such as Terraform.
```

The NFS server name is specified by the EFS FSID and the AWS region.

> Be aware also to set the EFS specific mount options.

Reviewer: ```suggestion
Ensure that the <<_efs_specific_nfs_mount_options,EFS specific mount options>> are also set.
```

[source,yaml]
----
parameters:
  nfs_subdir_external_provisioner:
    helm_values:
      nfs:
        server: fs-e<fsid>.efs.<region>.amazonaws.com
        path: /
----

=== EFS specific NFS mount options

> AWS recommends to use a [https://docs.aws.amazon.com/efs/latest/ug/mounting-fs-mount-cmd-general.html](specific set of NFS mount options) when using EFS NFS.

Reviewer: ```suggestion
AWS recommends to use a https://docs.aws.amazon.com/efs/latest/ug/mounting-fs-mount-cmd-general.html[specific set of NFS mount options] when using EFS NFS.
```
